openreview/venue/invitation.py

```diff
@@ -2187,9 +2187,14 @@ def set_submission_revision_invitation(self, submission_revision_stage=None):
         content = revision_stage.get_content(api_version='2', conference=self.venue)
 
         hidden_field_names = self.venue.submission_stage.get_hidden_field_names()
+        existing_invitation = tools.get_invitation(self.client, revision_invitation_id)
+        invitation_content = existing_invitation.edit['invitation']['edit']['note']['content'] if existing_invitation else {}
+
         for field in content:
             if field in hidden_field_names:
                 content[field]['readers'] = [venue_id, self.venue.get_authors_id('${{4/id}/number}')]
+            if field not in hidden_field_names and invitation_content.get(field, {}).get('readers', []):
```

Comment:
@melisabok I reverted back to adding readers to hidden fields. However, I also now check if the invitation exists and if a field has readers that shouldn't be hidden anymore, I remove the readers.

Reply:
This would prevent the issue we had with NAACL today.